src/lib/symbol_overrides.cc

```diff
@@ -92,10 +72,10 @@ struct malloc {
 
   static void *hook(size_t size) noexcept {
     check_libraries();
-    Guard guard(&g_in_allocator_guard);
+    ddprof::ReentryGuard guard(nullptr);
```

Comment:
I initially liked this because I did not have to retrieve the TLS storage from outside, however it does not work as I intended it to work (now it does the opposite of what the initial version was doing).